base/linalg/blas.jl

```diff
@@ -339,6 +369,15 @@ nrm2(x::Array) = nrm2(length(x), pointer(x), 1)
     asum(n, X, incx)
 
 Sum of the absolute values of the first `n` elements of array `X` with stride `incx`.
+
+# Example:
+```jldoctest
+julia> Base.BLAS.asum(10, im*ones(10), 2)
+5.0
```

Comment:
I think what it's doing is saying "the total length of the array is 10, within those ten, take steps of width 2". For example:

```julia
julia> a = [1; 2; 4; 6; 8; 10]
6-element Array{Int64,1}:
  1
  2
  4
  6
  8
 10

julia> Base.BLAS.asum(6, float(a), 1)
31.0

julia> Base.BLAS.asum(6, float(a), 2)
13.0

julia> Base.BLAS.asum(6, float(a), 4)
9.0
```
It's taking elements up to the max length of `6`.